### PR TITLE
Feature/3 ds 2

### PR DESCRIPTION
**Pull request by mreams-bambora**
Merge 3DS 2 docs into realtime-payments so they can be deployed and shared.

### CODE REVIEW

File: data/na/swagger/1-0-5.yaml [hunk 1]
```diff
@@ -1985,11 +2003,14 @@ definitions:
   CardResponse:
     description: 'For a 3D Secure payment process.'
     required:
-      - pa_res
+      - cres
     properties:
       pa_res:
         type: string
-        description: 'For a 3D Secure payment process this stores the Authentication Code (PaRes).'
+        description: 'For a 3D Secure payment process this stores the Authentication Code (PaRes).  NOTE: No longer used, this field is renamed to cres'.
```
Comment (Bambora-Dave-Robinson) [inline data/na/swagger/1-0-5.yaml]: Is this true? I don't know for sure but I had assumed 3DSv1 would remain untouched, which means this value is still relevant for 3DSv1

File: data/na/swagger/1-0-5.yaml [hunk 2]
```diff
@@ -2010,6 +2037,50 @@ definitions:
       eci:
         type: integer
         description: 'The electronic commerce indicator (ECI) used for processing transactions with 3D Secure.'
+  BrowserData:
+    description: |
+      Data collected about the cardholders browser, required for 3DS 2.0 transactions, optional for others.
+    required:
+      - acceiptHeader
```
Comment (Bambora-Dave-Robinson) [inline data/na/swagger/1-0-5.yaml]: Typo

File: data/na/swagger/1-0-5.yaml [hunk 3]
```diff
@@ -2010,6 +2037,50 @@ definitions:
...
+      - javaEnabled
+      - language
+      - screenHeight
+      - screenWidth
+      - timeZone
+      - userAgent
+      - JavascriptEnabled
```
Comment (Bambora-Dave-Robinson) [inline data/na/swagger/1-0-5.yaml]: latest version has this begin with a lowercase 'j' in order to be consistent

File: data/na/swagger/1-0-5.yaml [hunk 4]
```diff
@@ -2010,6 +2037,50 @@ definitions:
...
+    properties:
+      acceptHeader:
+        type: string
+        maxLength: 2048
+        description: 'Accept Header returned by the browser.'
```
Comment (Bambora-Dave-Robinson) [inline data/na/swagger/1-0-5.yaml]: This should mention that this is an 'HTTP' header

File: data/na/swagger/1-0-5.yaml [hunk 5]
```diff
@@ -2010,6 +2037,50 @@ definitions:
...
+      screenWidth:
+        type: integer
+        description: 'Width, in pixels, of the screen.'
+      timeZone:
+        type: integer
+        description: 'Timezone returned by the browser.'
+      userAgent:
+        type: string
+        maxLength: 2048
+        description: 'User-Agent header returned by the browser.'
```
Comment (Bambora-Dave-Robinson) [inline data/na/swagger/1-0-5.yaml]: This should mention that it comes in the client HTTP request, as it cannot be fetched in javascript like the other values

File: data/na/swagger/1-0-5.yaml [hunk 6]
```diff
@@ -2010,6 +2037,50 @@ definitions:
...
+      JavascriptEnabled:
```
Comment (Bambora-Dave-Robinson) [inline data/na/swagger/1-0-5.yaml]: start with a lowercase 'j'

File: source/docs/guides/3D_secure.md [hunk 7]
```diff
@@ -52,30 +52,43 @@ curl https://api.na.bambora.com/v1/payments \
...
+         "browser": {
+            "acceptHeader": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3",
+            "javaEnabled": "false",
+            "language": "en-US",
+            "colorDepth": "24bits",
+            "screenHeight": 1080,
+            "screenWidth": 1920,
+            "timeZone": -120,
+            "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36",
+            "JavascriptEnabled": true,
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/3D_secure.md]: Lowercase 'j'

File: source/docs/guides/3D_secure.md [hunk 8]
```diff
@@ -242,4 +263,51 @@ Response
...
+       - Browser data
+           - acceptHeader, string (1-2048), required
+           - javaEnabled, Boolean true/false, required
+           - language, string (1-8), required
+           - colorDepth, 
+           - screenHeight, numeric (0 to 9999999), required
+           - screenWidth, numeric (0 to 9999999), required
+           - timeZone, numeric (-840 to 720), required
+           - userAgent, string (1-2048), required
+           - JavascriptEnabled, optional Boolean true/false, default true.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/3D_secure.md]: Lowercase 'j'

File: source/includes/na/take_payments/_payment_3d_secure.md [hunk 9]
```diff
@@ -5,20 +5,37 @@ Definition
...
+         "browser": {
+            "acceptHeader": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3",
+            "javaEnabled": "false",
+            "language": "en-US",
+            "colorDepth": "24bits",
+            "screenHeight": 1080,
+            "screenWidth": 1920,
+            "timeZone": -120,
+            "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36",
+            "JavascriptEnabled": true,
```
Comment (Bambora-Dave-Robinson) [inline source/includes/na/take_payments/_payment_3d_secure.md]: Lowercase 'j'

File: source/docs/guides/3D_secure.md [hunk 10]
```diff
@@ -52,30 +52,43 @@ curl https://api.na.bambora.com/v1/payments \
...
+   "customer_ip": "123.123.123.123"
+   "token": {
+      "name": "Test User",
+      "code":"gt7-0f2f20dd-777e-487e-b688-940b526172cd",
+      "3d_secure": {
+         "browser": {
+            "acceptHeader": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3",
+            "javaEnabled": "false",
+            "language": "en-US",
+            "colorDepth": "24bits",
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/3D_secure.md]: No 'bits' on here

File: data/na/swagger/1-0-5.yaml [hunk 11]
```diff
@@ -1985,11 +2003,14 @@ definitions:
   CardResponse:
     description: 'For a 3D Secure payment process.'
     required:
-      - pa_res
+      - cres
```
Comment (Bambora-Dave-Robinson) [inline data/na/swagger/1-0-5.yaml:2007]: pa_res will be required for 3DSv1 still.

File: source/docs/guides/3D_secure.md [hunk 12]
```diff
@@ -52,40 +52,53 @@ curl https://api.na.bambora.com/v1/payments \
...
+
+   "amount": 250.01,
+   "payment_method":"token",
+   "customer_ip": "123.123.123.123"
+   "token": {
+      "name": "Test User",
+      "code":"gt7-0f2f20dd-777e-487e-b688-940b526172cd",
+      "3d_secure": {
+         "browser": {
+            "acceptHeader": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3",
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/3D_secure.md]: These need to be updated with the underscore as well

File: source/docs/guides/3D_secure.md [hunk 13]
```diff
@@ -52,40 +52,53 @@ curl https://api.na.bambora.com/v1/payments \
...
+ "href":"https://api.na.bambora.com/v1/payments/2ccd7715-9e97-4f11-9fff36e6584e3200/continue","method":"POST"
+ }
+ ]
 }
 ```
 
 In the 302 response above, the 'merchant_data' attribute value should be saved in the current users session.
 
-The merchant’s process URL decodes the response redirect and displays the information in the customer’s web browser. 
+The merchantâ€™s process URL decodes the response redirect and displays the information in the customerâ€™s web browser.
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/3D_secure.md]: Something odd about the apostrophe here

File: source/docs/guides/3D_secure.md [hunk 14]
```diff
@@ -177,7 +198,7 @@ system variables:
 | Attribute | Description |
 | --- | --- |
 | xid | Include the 3D Secure transaction identifier (up to 20-digits). |
-| eci | SecureECI is a 1-digit status code: 5 – authenticated; 6 – attempted, not completed. |
+| eci | SecureECI is a 1-digit status code: 5 â€“ authenticated; 6 â€“ attempted, not completed. |
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/3D_secure.md]: Dashes not showing in browser properly

File: source/docs/guides/3D_secure.md [hunk 15]
```diff
@@ -242,4 +263,51 @@ Response
...
+along with new optional parameters to indicate the 3DS version and if a transaction should continue to be completed if 3DS authentication fails.
+
+The payment response will no longer be retuning an eci code to indicate if 3DS authentication was successful where will instead return a status code
+and in the case of an failure, an error code.  The naming of the parameters in the redirection will also be changing, but their contents and usage will be the same.
+
+##Detailed list of changes
+
+    - Payment API request
+       - Browser data
+           - acceptHeader, string (1-2048), required
```
Comment (Bambora-Dave-Robinson) [inline source/docs/guides/3D_secure.md]: Underscore

File: source/includes/na/take_payments/_payment_3d_secure.md [hunk 16]
```diff
@@ -5,20 +5,37 @@ Definition
...
+   "payment_method": "card",
+   "customer_ip": "123.123.123.123"
+   "card": {
+      "name": "Test User",
+      "number": "4012000033330026",
+      "expiry_month": "09",
+      "expiry_year": "20",
+      "3d_secure": {
+         "browser": {
+            "acceptHeader": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3",
```
Comment (Bambora-Dave-Robinson) [inline source/includes/na/take_payments/_payment_3d_secure.md]: Underscore

File: source/docs/guides/3D_secure.md (hunk 12 above)
Comment (mreams-bambora) [reply source/docs/guides/3D_secure.md]: doh! fixed now.

File: source/docs/guides/3D_secure.md (hunk 14 above)
Comment (mreams-bambora) [reply source/docs/guides/3D_secure.md]: Oh shoot those were supposed to be dashes!

File: source/docs/guides/3D_secure.md (hunk 13 above)
Comment (mreams-bambora) [reply source/docs/guides/3D_secure.md]: This time I think I got all of them.

**Comment by Lmasniuk** [comment]
Looks good to me! Ship it!

All jokes aside, might want to remove me from this repo. Hope all is well :D <3

**Review by Bambora-Dave-Robinson — APPROVED** [review]

**Pull request merged**